test/autests/gold_tests/field_verification/substr_rules.test.py

```diff
@@ -0,0 +1,114 @@
+
+client.ReturnCode = 1
+server.ReturnCode = 1
+
+#
+# Test 2: Verify duplicate field verification in a YAML replay file.
+#
+r = Test.AddTestRun("Verify field verification works for HTTP transaction with duplicate fields")
+client = r.AddClientProcess("client2", "http_replay_files/substr_rules_duplicate", http_ports=[8080], other_args="--verbose diag")
+server = r.AddServerProcess("server2", "http_replay_files/substr_rules_duplicate", http_ports=[8081], other_args="--verbose diag")
+proxy = r.AddProxyProcess("proxy2", listen_port=8080, server_port=8081)
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Contains Violation: Absent/Mismatched. Key: "1", Name: "set-cookie", Required Values: "AB", Received Values: "ABCD" "EFG"',
+        'Validation should be complain that "Set-Cookie" had too many values.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Prefix Violation: Absent/Mismatched. Key: "1", Name: "pref-cookie", Required Prefixes: "AB" "EF", Received Values: "ABCD"',
+        'Validation should be complain that "Set-Cookie" had too few values.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Suffix Violation: Absent/Mismatched. Key: "1", Name: "suff-cookie", Required Suffixes: "AB" "EF", Received Values:',
+        'Validation should complain that "Set-Cookie" had no values.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Contains Violation: Not Contained. Key: "1", Name: "set-cookie", Required Values: "G" "F", Received Values: "ABCD" "EFG"',
+        'Validation should be complain that "Set-Cookie" had uncontained values.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Prefix Violation: Not Found. Key: "1", Name: "set-cookie", Required Prefixes: "BC" "EF", Received Values: "ABCD" "EFG"',
+        'Validation should be complain that "Set-Cookie" did not match all prefixes.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Suffix Violation: Not Found. Key: "1", Name: "set-cookie", Required Suffixes: "AB" "G", Received Values: "ABCD" "EFG"',
+        'Validation should complain that "Set-Cookie" did not match all suffixes.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Contains Success: Key: "1", Name: "set-cookie", Required Values: "AB" "E", Received Values: "ABCD" "EFG"',
+        'Validation should be happy that "Set-Cookie" matched all values.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Prefix Success: Key: "1", Name: "set-cookie", Required Prefixes: "A" "EFG", Received Values: "ABCD" "EFG"',
+        'Validation should be happy that "Set-Cookie" matched all prefixes.')
+
+server.Streams.stdout += Testers.ContainsExpression(
+        'Suffix Success: Key: "1", Name: "set-cookie", Required Suffixes: "ABCD" "EFG", Received Values: "ABCD" "EFG"',
+        'Validation should be happy that "Set-Cookie" matched all suffixes.')
+
+server.ReturnCode = 1
```

Comment:
Hmm. Interesting. github says this is missing a terminating newline. Can you please add that?